adserver/analyzer/backends/textacynlp.py

```diff
@@ -0,0 +1,95 @@
+"""Textacy/Spacy-based keyword analyzer that lemmatizes and uses textrank for keyphrase extraction."""
+import collections
+
+import textacy
+from textacy import preprocessing
+
+from .naive import NaiveKeywordAnalyzerBackend
+
+
+class TextacyAnalyzerBackend(NaiveKeywordAnalyzerBackend):
+
+    """
+    An analyzer backend that uses textacy for lemmatization and keyword extraction.
+
+    https://textacy.readthedocs.io/en/latest/quickstart.html
+    """
+
+    TOP_PHRASE_COUNT = 20
+
+    # Minimum phrase length where each word isn't required to be in the output phrase
+    MIN_PHRASE_LENGTH = 6
+
+    def __init__(self, url, **kwargs):
+        """Overrides to lemmatize keywords."""
+        super().__init__(url, **kwargs)
+
+        self.pretrained_model = textacy.load_spacy_lang(
+            "en_core_web_md", disable=("parser",)
+        )
+
+        self.preprocessor = preprocessing.make_pipeline(
+            preprocessing.normalize.unicode,
+            preprocessing.remove.punctuation,
+            preprocessing.normalize.whitespace,
+        )
+
+        # Maps lemmatized keywords back to the original keyword/phrase in our database
+        self.lemmatized_to_original_map = {}
+
+        # Reset the keywords
+        self.keyword_corpus = []
+
+        # Build up the lemmatized keywords
+        for topic in self.topics:
+            for keyword in self.topics[topic]:
+                terms_doc = textacy.make_spacy_doc(
+                    self.preprocessor(keyword).lower(), lang=self.pretrained_model
+                )
+                words = list(
```

Comment:
So we're basically pre-computing the lemma for every keyword here? Would be curious what this actually looks like, I'll plan to drop into a PDB locally and step through it.